janusgraph-core/src/main/java/org/janusgraph/graphdb/query/vertex/BasicVertexCentricQueryBuilder.java

```diff
@@ -418,6 +420,20 @@ protected BaseVertexCentricQuery constructQueryWithoutProfile(RelationCategory r
         assert returnType != null;
         Preconditions.checkArgument(adjacentVertex==null || returnType == RelationCategory.EDGE,
                 "Vertex constraints only apply to edges");
+
+        if(this.adjacentVertex != null && this.types == NO_TYPES){
+            Iterable<JanusGraphVertex> allTypes = QueryUtil.getVertices(this.tx,
+                    BaseKey.SchemaCategory, JanusGraphSchemaCategory.EDGELABEL);
+            int length = IterablesUtil.size(allTypes);
+            int index = 0;
+            String[] labels = new String[length];
+            for(JanusGraphVertex labelVertex : allTypes){
+                labels[index] = ((EdgeLabel)labelVertex).name();
+                index++;
+            }
+            labels(labels);
```

Comment:
This seems to be a working fix, but it's more like a quick workaround. I think we need to find the root cause first.

What I have observed is, when Direction is BOTH, the bug appears, but when Direction is IN or OUT, the bug does not appear. So the root cause might be related to that.